docs/vector_search.rst

```diff
@@ -0,0 +1,278 @@
+Vector Search and Embeddings in Nrtsearch
+==========================================
+
+What is Vector/Embeddings?
+--------------------------
+
+Vector/embeddings are a simple data structure consisting of a list of (float) numbers with N dimensions to represent a variety of complex data such as photos, videos, textual data, etc., that cannot be stored in traditional databases or datastores in a searchable way. By utilizing this data structure, search engines that support vector search can find similar documents using KNN (K nearest neighbor) or ANN (approximate nearest neighbor) algorithms.
+
+For example, one can search for photos using textual descriptions such as *"vegetarian pizza."* To achieve this, we need to first convert a large number of photos into vector embeddings, such as::
+
+    [[1.0, 0.3, 2.1], [0.01, 0.02, 1.24], ...]
+
+using an appropriate algorithm and store them in a database that supports vector search. Subsequently, users' textual queries will be transformed into vector embeddings, such as::
+
+    [1.0, 3.2, 1.3]
+
+The algorithm will then determine the most relevant photos by traversing through a graph and applying similarity algorithms such as cosine similarity, dot product, or others.
+
+You can find more detailed information about vector embeddings in the following articles:
+
+- `Google <https://cloud.google.com/blog/topics/developers-practitioners/meet-ais-multitool-vector-embeddings>`_
+- `Elastic <https://www.elastic.co/what-is/vector-embedding>`_
+
+Vector Search Applications
+--------------------------
+
+The following is a list of example applications that can benefit from vector search:
+
+- Photo search
+- Review search
+- Similar businesses search
+- Bot detection
+- Chatbots
+
+Vector Embeddings on Nrtsearch
+------------------------------
+
+Nrtsearch is based on the popular open-source search library, Lucene. Vector search support was added to Lucene in version 9+. It uses the Hierarchical Navigable Small Worlds (HNSW) graph to traverse vector data to find the most relevant results. It supports vector-only search as well as hybrid (vector + text search).
+
+More information about Lucene's support of vector search can be found at:
+
+- `Lucene Vector Search <https://www.apachecon.com/acna2022/slides/04_lucene_vector_search_sokolov.pdf>`_
+
+In the following sections, we will go through different steps to launch and configure an Nrtsearch cluster with vector search support.
+
+What was Already Supported?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Nrtsearch v0 already supports regular vector fields. However, it's not using any kind of graph data structure under the hood to do advanced KNN lookups in the recall/boosting phase. This data can only be used in the scoring phase to find similarity scores using custom functions/painless scripts.
```

Comment:
I'm not sure the point of this section. Discussing v0 support is a bit confusing.

Reply:
Removed this